lib/ProductOpener/Export.pm

```diff
@@ -360,29 +408,92 @@ sub export_csv($) {
 
 	# Second pass - output CSV data
 
-	my $csv = Text::CSV->new ( { binary => 1 , sep_char => $separator } )  # should set binary attribute.
-		or die "Cannot use CSV: ".Text::CSV->error_diag ();
+	# Initialize the XLSX or CSV modules, optionally send HTTP headers (if called from the website)
+	# and output the header row
+	my $csv;
+	my $workbook;
+	my $worksheet;
+
+	if ($format eq "xlsx") {
+
+		# Send HTTP headers, unless search_and_export_products() is called from a script
+		if ($send_http_headers) {
+			require Apache2::RequestRec;
+			my $r = Apache2::RequestUtil->request();
+			$r->headers_out->set("Content-type" => "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
+			$r->headers_out->set("Content-disposition" => "attachment;filename=$filename");
+			print "Content-Type: application/vnd.openxmlformats-officedocument.spreadsheetml.sheet\r\n\r\n";
+
+			binmode($filehandle);
+		}
```

Comment:
Not mandatory, but I would really advocate to put it in a separate function which is called where needed, by caller of `export_csv`

Deep nesting web processing deep inside the code is bad practice. (remember I spoke about de-mangling our code)

Reply:
Agreed, but I can't think of a good way to do it, because the caller of export_csv does not know in advance if it will have to show an error (e.g. database issue), or if the export will succeed.